File: yh_sum/src/yh_sum_sub.cpp
#include "ros/ros.h"
#include "yh_sum/yh_sum_msg.h"


int old_data = 0;

void msgCallback(const yh_sum::yh_sum_msg::ConstPtr& msg)
{
    printf("%d\n",old_data + msg->data);
    old_data = msg->data;

}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "yh_sum_sub");
    ros::NodeHandle nh;

    ros::Subscriber sub =  nh.subscribe("yh_sum_topic", 100, msgCallback);
    ros::spin();  //프로그램 종료 방지 대기

    return 0;

}

File: yh_sum/scripts/yh_sum_sub.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

import rospy
from yh_sum.msg import yh_sum_msg

old_data = 0
def msgCallback(msg):
    global old_data
    print(old_data + msg.data)
    old_data = msg.data
    

def listner():
    rospy.init_node("yh_sum_sub", anonymous=True)
    rospy.Subscriber("yh_sum_topic", yh_sum_msg, msgCallback)

    rospy.spin()

if __name__ == "__main__":
    listner()